src/open_clip/model.py

```diff
@@ -228,15 +228,35 @@ def encode_image(self, image, normalize: bool = False):
     def encode_text(self, text, normalize: bool = False):
         cast_dtype = self.transformer.get_cast_dtype()
 
-        x = self.token_embedding(text).to(cast_dtype)  # [batch_size, n_ctx, d_model]
```

Comment:
Better move all of these modifications to a new class that inherits from this class.

Reply:
is this really necessary? it would mean a lot of changes throughout the openclip repo